src/core/server/plugins/plugins_service.test.ts

```diff
@@ -1065,6 +1065,68 @@ describe('PluginsService', () => {
     });
   });
 
+  test('"root" deprecations from one plugin should be applied before accessing other plugins config', async () => {
+    const pluginA = createPlugin('plugin-1-deprecations', {
+      type: PluginType.standard,
+      path: 'plugin-1-deprecations',
+      version: 'version-1',
+    });
+
+    const pluginB = createPlugin('plugin-2-deprecations', {
+      type: PluginType.standard,
+      path: 'plugin-2-deprecations',
+      version: 'version-2',
+    });
+
+    jest.doMock(
+      join(pluginA.path, 'server'),
+      () => ({
+        config: {
+          schema: schema.object({
+            enabled: schema.maybe(schema.boolean({ defaultValue: true })),
+          }),
+        },
+      }),
+      { virtual: true }
+    );
+
+    jest.doMock(
+      join(pluginB.path, 'server'),
+      () => ({
+        config: {
+          schema: schema.object({
+            enabled: schema.maybe(schema.boolean({ defaultValue: true })),
+            renamed: schema.string(), // Mandatory string to make sure that the field is actually renamed by deprecations
+          }),
+          deprecations: (toolkit: any) => [
```

Comment:
NIT: I would cast the whole object as `PluginConfigDescriptor` or the deprecations as `ConfigDeprecationProvider` to avoid using `toolkit: any`